src/morphological_operations.jl

```diff
@@ -99,3 +99,70 @@ which is defined as the arithmetic difference between the internal and the exter
 `region` allows you to control the dimensions over which this operation is performed.
 """
 morpholaplace(img::AbstractArray, region=coords_spatial(img)) = dilate(img, region) + erode(img, region) - 2img
+
+
+"""
+```
+function thinning_iteration(img::AbstractArray{Bool,2}; odd_iteration::Bool)
+```
+The thining iteration evaluates three conditions in order to determine which pixels of the image should be removed from img.
+The three conditions are explained in:
+Guo, Z., & Hall, R. W. (1989). Parallel thinning with two-subiteration algorithms. Communications of the ACM, 32(3), 359-373.
+"""
+function thinning_iteration!(img_ori::AbstractArray{Bool,2}, odd_iteration::Bool)
+    marker = trues(size(img_ori))
+    img = falses(size(img_ori).+2)
+    img[2:end-1,2:end-1] = img_ori
+    h, w = size(img)
+    for j=2:w-1, i=2:h-1
+        if !img[i,j]
+            continue
+        end
+        p1 = img[i-1,j-1]
+        p2 = img[i-1,j]
+        p3 = img[i-1,j+1]
+        p4 = img[i,j+1]
+        p5 = img[i+1,j+1]
+        p6 = img[i+1,j]
+        p7 = img[i+1,j-1]
+        p8 = img[i,j-1]
+        A = (!p2 && (p3 || p4)) + (!p4 && (p5 || p6)) + (!p6 && (p7 || p8)) + (!p8 && (p1 || p2))
+        B1 = (p1 || p2) + (p3 || p4) + (p5 || p6) + (p7 || p8)
+        B2 = (p2 || p3) + (p4 || p5) + (p6 || p7) + (p8 || p1)
+        B = min(B1, B2)
+        G3 = false
+        if odd_iteration
+            G3 = (p2 || p3 || (!p5)) && p4
+        else
+            G3 = (p6 || p7 || (!p1)) && p8
+        end
+        if (A == 1) && ((2 <= B) && (B <= 3)) && (!G3) 
+            marker[i-1,j-1] = false
+        end
+    end
+    img_ori .= img_ori .& marker
+end
+
+doc"""
+```julia
+function thinning(img::AbstractArray{Bool})
+```
+
+Applies a binary blob thinning operation, to achieve a skeletization of the input image.
+The procedure carries on until there are no changes between two consecutives iterations.
+
+The algorithm is described in:
+1. Guo, Z., & Hall, R. W. (1989). Parallel thinning with two-subiteration algorithms. Communications of the ACM, 32(3), 359-373.
+2. Lam, L., Lee, S. W., & Suen, C. Y. (1992). Thinning methodologies-a comprehensive survey. IEEE Transactions on pattern analysis and machine intelligence, 14(9), 869-885.
+"""
+function thinning(img::AbstractArray{Bool}) 
```

Comment:
It would be nice to allow future contributors to implement other thinning algorithms:

```julia
function thinning(img::AbstractArray{Bool}; algo::ThinAlgo=GuoAlgo())
    # dispatch appropriate implementation
    thinning_impl(img, algo)
end
```

The function would be dispatched on the algorithm type `GuoAlgo` or any other more informative name from the paper. The type could be empty like:

```julia
abstract type ThinAlgo end
struct GuoAlgo <: ThinAlgo end
```

since this specific algorithm doesn't have tuning parameters.

Reply:
I forgot to mention that after I added the code to process the image border I couldn't find differences with  matlab implementation anymore. I will add nevertheless the pages where the algorithm is explained.
I like the idea to have an algorithm type to dispatch on. But I have a question:: Where should I put the documentation of the method. ¿In the concrete type or in the function?

Reply:
That is awesome, good to know that the difference is gone. :) Maybe we should add the doc string to the type? What do you think? The general function could point to the individual doc strings like a "See also" section.